src/Merchants.php

```
@@ -0,0 +1,90 @@
+
+		$merchant          = false;
+		$merchant_id       = Pinterest_For_Woocommerce()::get_data( 'merchant_id' );
+		$saved_merchant_id = $merchant_id;
+
+		if ( empty( $merchant_id ) ) {
+			// Get merchant from advertiser object.
+			$merchant_id = self::get_merchant_id_from_advertiser();
+		}
+
+		if ( ! empty( $merchant_id ) ) {
+
+			try {
+				$merchant = API\Base::get_merchant( $merchant_id );
+				if ( $saved_merchant_id !== $merchant_id ) {
+					Pinterest_For_Woocommerce()::save_data( 'merchant_id', $merchant['data']->id );
+				}
+			} catch ( \Throwable $th ) {
+				$merchant = false;
+			}
+		}
+
+		if ( ! $merchant || ( 'success' !== $merchant['status'] && 650 === $merchant['code'] ) ) {  // https://developers.pinterest.com/docs/redoc/#tag/API-Response-Codes Merchant not found 650.
+			// Try creating one.
+			$merchant = API\Base::maybe_create_merchant( $feed_args );
+			if ( 'success' === $merchant['status'] ) {
+				Pinterest_For_Woocommerce()::save_data( 'merchant_id', $merchant['data']->id );
+			}
+		}
+
+		if ( ! $merchant || 'success' !== $merchant['status'] ) {
+			throw new \Exception( esc_html__( 'Response error when trying create a merchant or get the existing one.', 'pinterest-for-woocommerce' ), 400 );
+		}
+
+		return $merchant;
+	}
+
+
+	/**
+	 * Gets the merchant ID of the authenticated user from the data returned on the Advertisers endpoint.
+	 *
+	 * @return string|null
+	 *
+	 * @throws \Exception PHP exception.
+	 */
+	private static function get_merchant_id_from_advertiser() {
+		$advertisers = API\Base::get_advertisers();
+
+		if ( 'success' !== $advertisers['status'] ) {
+			throw new \Exception( esc_html__( 'Response error when trying to get advertisers.', 'pinterest-for-woocommerce' ), 400 );
+		}
+
+		$advertiser = reset( $advertisers['data'] ); // All advertisers assigned to a user share the same merchant_id.
+
+		return ! empty( $advertiser->merchant_id ) ? $advertiser->merchant_id : null;
```

Instead of returning `null`, it would be better to throw an `Exception` and allow the calling function to catch that exception. Then this function only has one type of return value.